app/src/main/java/com/keylesspalace/tusky/components/trending/TrendingLinkViewHolder.kt

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2023 Tusky Contributors
+ *
+ * This file is a part of Tusky.
+ *
+ * This program is free software; you can redistribute it and/or modify it under the terms of the
+ * GNU General Public License as published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Tusky is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+ * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+ * Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with Tusky; if not,
+ * see <http://www.gnu.org/licenses>.
+ */
+
+package com.keylesspalace.tusky.components.trending
+
+import android.view.ViewGroup.LayoutParams.MATCH_PARENT
+import android.view.ViewGroup.LayoutParams.WRAP_CONTENT
+import android.widget.ImageView.ScaleType
+import android.widget.LinearLayout
+import androidx.recyclerview.widget.RecyclerView
+import com.bumptech.glide.Glide
+import com.google.android.material.shape.CornerFamily
+import com.google.android.material.shape.ShapeAppearanceModel
+import com.keylesspalace.tusky.R
+import com.keylesspalace.tusky.databinding.ItemTrendingLinkBinding
+import com.keylesspalace.tusky.entity.TrendsLink
+import com.keylesspalace.tusky.util.StatusDisplayOptions
+import com.keylesspalace.tusky.util.decodeBlurHash
+import com.keylesspalace.tusky.util.hide
+
+class TrendingLinkViewHolder(
+    private val binding: ItemTrendingLinkBinding,
+    private val onClick: (String) -> Unit
+) : RecyclerView.ViewHolder(binding.root) {
+    private var link: TrendsLink? = null
+
+    private val radius = binding.cardImage.resources.getDimensionPixelSize(R.dimen.card_radius).toFloat()
+
+    init {
+        itemView.setOnClickListener { link?.let { onClick(it.url) } }
+    }
+
+    fun bind(link: TrendsLink, statusDisplayOptions: StatusDisplayOptions) {
+        this.link = link
+
+        // TODO: This is very similar to the code in StatusBaseViewHolder.setupCard().
+        // Can a "PreviewCardView" type be created to encapsulate all this?
+        binding.cardTitle.text = link.title
+
+        val description = if (link.description.isNotBlank()) {
+            link.description
+        } else if (link.authorName.isNotBlank()) {
+            link.authorName
+        } else {
+            null
+        }
+        description?.let { binding.cardDescription.text = it } ?: binding.cardDescription.hide()
+
+        binding.cardLink.text = link.url
+
+        val cardImageShape = ShapeAppearanceModel.Builder()
+
+        if (statusDisplayOptions.mediaPreviewEnabled && !link.image.isNullOrBlank()) {
```

Comment:
What does all this do (comment?)?

It seems to adapt layout dynamically but to what end?

Reply:
There are two preview layout types.

One puts the preview image above the link title and description (e.g., see the screenshots in the second comment on this PR).

The other puts the preview to the left of the title and description. As the previous comment notes, this is out of StatusBaseViewHolder, and might be moved somewhere common later if it needs reusing again.

This is to follow what I call the rule-of-three for refactoring (see also https://en.wikipedia.org/wiki/Don%27t_repeat_yourself#AHA).

1. You don't need to refactor code that's only used once into a common function, because you don't have the necessary knowledge to know what parts of the code should be governed by parameters (you have zero examples of reuse).
2. You don't need to refactor code that's only used twice in to a common function, because you still don't have the necessary knowledge to know what parts of the code should be governed by parameters (you have one example of reuse, which is not enough).
3. When you've copy pasted the same code in to the three different places... now is the time to start looking at refactoring in to common function, because you've three distinct use cases that you can learn from (you have two examples of reuse, which is starting to become useful).

Reply:
One way to see the other layout is to disable downloading media, then it looks like this:

<img src="https://github.com/tuskyapp/Tusky/assets/773100/68ec3987-88bf-4fe7-89ef-0ea9c24b50e5" width="45%">

Reply:
> There are two preview layout types.
> 
> One puts the preview image above the link title and description (e.g., see the screenshots in the second comment on this PR).
> 
> The other puts the preview to the left of the title and description. As the previous comment notes, this is out of StatusBaseViewHolder, and might be moved somewhere common later if it needs reusing again.

These might be the code comment(s) I was looking for.